utopia-vscode-common/src/mailbox.ts

```diff
@@ -70,27 +70,44 @@ async function initOutbox(outboxToUse: Mailbox): Promise<void> {
   }
 }
 
-async function receiveMessage<T>(messageName: string, parseMessage: (msg: string) => T): Promise<T> {
+async function receiveMessage<T>(
+  messageName: string,
+  parseMessage: (msg: string) => T,
+): Promise<T> {
   const messagePath = pathForInboxMessage(messageName)
   const content = await readFileSavedContentAsUTF8(messagePath)
   return parseMessage(content)
 }
 
+async function waitForPathToExist(path: string, maxWaitTime: number = 5000): Promise<void> {
+  if (maxWaitTime >= 0) {
+    const doesItExist: boolean = await exists(path)
+    if (!doesItExist) {
+      await waitForPathToExist(path, maxWaitTime - 100)
+    }
+  } else {
+    return Promise.reject(`Waited too long for ${path} to exist.`)
```

Comment:
I _think_ we have a convention to use `throw` to return a failed promise in async functions

Reply:
We've used `Promise.reject` in quite a few other places in the `utopia-vscode-common` code. We did hit a very bizarre issue when using `throw` inside promises when first experimenting with the VSCode integration - some sequence of events was causing the `throw` to be handled outside of the promise handling code (even though debugging showed it to clearly be thrown from inside the promise correctly), leading to it being treated as an unhandled exception that would break the extension. We never did get to the bottom of that...

Reply:
If it's going to potentially breach the Promise contract, I'd definitely much rather go with `Promise.reject` unless there's a strong reason otherwise.